peer.js

```diff
@@ -410,6 +417,47 @@ TChannelPeer.prototype.connectTo = function connectTo() {
     return self.connect(true);
 };
 
+TChannelPeer.prototype.tryConnect = function tryConnect() {
+    var self = this;
+
+    var connectTime = Date.now();
+    if (connectTime < self.nextConnAttemptTime) {
+        return;
+    }
+
+    var conn = this.getOutConnection();
+    if (!conn || conn.direction !== 'out') {
+        conn = this.connectTo();
+    }
+
+    this.waitForIdentified(conn, onIdentified);
+
+    function onIdentified(err) {
+        if (!err) {
+            self.nextConnAttemptDelay = 0;
+            self.nextConnAttemptTime = 0;
+            return;
+        }
+
+        if (self.nextConnAttemptDelay === 0) {
+            self.nextConnAttemptDelay = INITIAL_CONN_ATTEMPT_DELAY;
+        } else {
+            self.nextConnAttemptDelay *= CONN_ATTEMPT_DELAY_MULTIPLER;
+            // Add some amount of fuzz, +-100ms
+            self.nextConnAttemptDelay += Math.floor(100 * (Math.random() - 0.5));
```

Comment:
oh ic, you're doing +/- 50ms... usually you'd just `x *= (1 + P * (Math.random() - 0.5))` for say P = 0.1 to get +/- 5% (aka 10% fuzz)...